app/models/repo.js

```diff
@@ -80,12 +97,16 @@ const Repo = VcsEntity.extend({
     const allowance = this.allowance;
     const isPrivate = this.private;
 
-    if (allowance && allowance.subscription_type === 1)
+    if (allowance && allowance.subscriptionType === 1)
       return true;
-    if (!allowance)
+    if (!allowance && !this.repoOwnerAllowance) {
+      return true; // pending allowance call
```

Comment:
It seems to me that this should return `false`. Until we know exactly that user is allowed to build, we shouldn't allow them to do so. Let's say the allowance request hung for a while and user had a chance to trigger a build (for example, via Restart) - they will likely get an error from API. If it's necessary to have this data already fetched at this point, you can trigger the task run in corresponding route - `beforeModel`, `model` and `afterModel` hooks are blocking view rendering until their returned promises are resolved.
Does it make sense to you?

Reply:
great hint! `beforeModel` seems to be created for such purposes

Reply:
Yes, it definitely is :)